runtime/bcverify/rtverify.c

```diff
@@ -1165,15 +1165,57 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 				break;
 
 			case JBreturn1:
+			case JBreturnB:
+			case JBreturnC:
+			case JBreturnS:
+			case JBreturnZ:
 			case JBsyncReturn1:
 				if (returnChar) {
+					U_32 returnType = 0;
 					/* single character return description */
 					if ((returnChar < 'A') || (returnChar > 'Z')) {
 						inconsistentStack = TRUE;
 						goto _illegalPrimitiveReturn;
 					}
 					type = (UDATA) oneArgTypeCharConversion[returnChar - 'A'];
 					POP_TOS(type);
+
+					/* check methods that return char, byte, short, or bool use the right opcode */
+					if (BCV_BASE_TYPE_INT == type){
+						switch(bc){
+						case JBreturnB:
+							returnType = BCV_BASE_TYPE_BYTE_BIT;
+							break;
+
+						case JBreturnC:
+							returnType = BCV_BASE_TYPE_CHAR_BIT;
+							break;
+
+						case JBreturnS:
+							returnType = BCV_BASE_TYPE_SHORT_BIT;
+							break;
+
+						case JBreturnZ:
+							returnType = BCV_BASE_TYPE_BOOL_BIT;
+							break;
+
+						case JBreturn1:
+						case JBsyncReturn1:
```

Comment:
Can you add a comment here that the synchronized {b,c,s,z} will always be handled by `JBgenericReturn`?  It will make it more obvious to anyone debugging through the verifier that this is correct without requiring them to check how the bytecodes are written